### PR TITLE
Updated docstring for get_cpds method in BayesianNetwork

### LINKED ISSUE #933: Correct docstring of get_cpds of BayesianModels
The docstring of `get_cpds` in `BayesianModel` doesn't specify that it returns a TabularCPD object when `node` is not `None. Specify this in the docstrings and also need to improve doctests.

### PR DESCRIPTION
**Pull request by B3CODER**
### Your checklist for this pull request
Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

 Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
 Make sure you are making a pull request against the dev branch (left side). Also you should start your branch off our dev.
 Check the commit's or even all commits' message styles matches our requested structure.

- Fixes #933

### List of changes to the codebase in this pull request
- Updated the docstring for the get_cpds method in BayesianNetwork to clarify the method's behavior when the node argument is specified and when it is not.
- Added more detailed descriptions to improve the documentation's clarity and usefulness for users.

### CODE REVIEW

File: pgmpy/models/BayesianNetwork.py [hunk 1]
```diff
@@ -279,8 +279,17 @@ def get_cpds(self, node=None):
 
         Returns
         -------
-        A list of TabularCPDs: list
-
+        - If 'node' is specified:
```
Comment (ankurankan) [inline pgmpy/models/BayesianNetwork.py]: I would keep the first line in the form of `name : type` form. Have a look at the numpydoc style for details: https://numpydoc.readthedocs.io/en/latest/format.html#returns

File: pgmpy/models/BayesianNetwork.py [hunk 2]
```diff
@@ -279,8 +279,17 @@ def get_cpds(self, node=None):
...
+            Returns a list of all TabularCPD objects that have been added to the model.
+        Raises
+        ------
+        ValueError
+            If the specified node is not present in the model.
+            
+            
         Examples
         --------
         >>> from pgmpy.models import BayesianNetwork
```
Comment (ankurankan) [inline pgmpy/models/BayesianNetwork.py]: I would completely switch out this example with something similar to the `get_example_model` example I had added to the issue.

**Review by ankurankan — CHANGES_REQUESTED** [review]
@B3CODER Thanks for creating the PR. I have added a few small comments to the changes.

File: pgmpy/models/BayesianNetwork.py [hunk 3]
```diff
@@ -291,6 +299,13 @@ def get_cpds(self, node=None):
         ...                  ['intel', 'diff'], [2, 2])
         >>> student.add_cpds(cpd)
         >>> student.get_cpds()
+
```
Comment (ankurankan) [inline pgmpy/models/BayesianNetwork.py]: Could you please update the examples here such that we use one of the example models? If you search for `get_example_model`, you will find plenty of docstring examples using the `get_example_model` method to create a model.

**Review by ankurankan — CHANGES_REQUESTED** [review]

**Comment by B3CODER** [comment]
### Changes Made:
- Updated the `get_cpds` method to provide consistent output for both the list of all CPDs and the CPD of a specific node.
- Added error handling for nodes not present in the model.

### Screenshot of the modified code:
image

File: pgmpy/models/BayesianNetwork.py [hunk 4]
```diff
@@ -279,18 +279,35 @@ def get_cpds(self, node=None):
...
         >>> from pgmpy.models import BayesianNetwork
-        >>> from pgmpy.factors.discrete import TabularCPD
-        >>> student = BayesianNetwork([('diff', 'grade'), ('intel', 'grade')])
-        >>> cpd = TabularCPD('grade', 2, [[0.1, 0.9, 0.2, 0.7],
-        ...                               [0.9, 0.1, 0.8, 0.3]],
-        ...                  ['intel', 'diff'], [2, 2])
-        >>> student.add_cpds(cpd)
-        >>> student.get_cpds()
+        >>> import pprint
+        >>> random_model = BayesianNetwork.get_random(n_nodes=8, edge_prob=0.4, n_states=2, latents=True)
```
Comment (ankurankan) [inline pgmpy/models/BayesianNetwork.py]: @B3CODER Here, you are currently using the `get_random` method to generate the model. I was suggesting to use the `get_example_model` method to generate the model. Something like this:
```python
from pgmpy.utils import get_example_model
model = get_example_model('cancer')
```
And then show examples similar to the ones that you have shown here.

File: pgmpy/models/BayesianNetwork.py [hunk 5]
```diff
@@ -608,7 +625,7 @@ def fit_update(self, data, n_prev_samples=None, n_jobs=1):
         --------
         >>> from pgmpy.utils import get_example_model
         >>> from pgmpy.sampling import BayesianModelSampling
-        >>> model = get_example_model('alarm')
+        >>> model = ('alarm')
```
Comment (ankurankan) [inline pgmpy/models/BayesianNetwork.py]: You would need to revert this.

File: pgmpy/utils/utils.py [hunk 6]
```diff
@@ -63,7 +63,7 @@ def get_example_model(model):
     Example
     -------
     >>> from pgmpy.data import get_example_model
-    >>> model = get_example_model(model='asia')
+    >>> model = (model='asia')
```
Comment (ankurankan) [inline pgmpy/utils/utils.py]: This as well.

**Review by ankurankan — CHANGES_REQUESTED** [review]

**Comment by ankurankan** [comment]
@B3CODER 

> Updated the get_cpds method to provide consistent output for both the list of all CPDs and the CPD of a specific node.
> Added error handling for nodes not present in the model.

I do not see any changes related to these in the PR. Have you pushed your local changes? Also, I would not make those changes as it would break a lot of other parts of the codebase.

**Comment by ankurankan** [comment]
@B3CODER The code formatting changes were needed only for the files that you had changes in, not for the whole repo. You can revert these by running:
```
git revert d31ba54
```

File: pgmpy/models/BayesianNetwork.py [hunk 7]
```diff
@@ -279,26 +279,45 @@ def get_cpds(self, node=None):
...
-        >>> from pgmpy.factors.discrete import TabularCPD
-        >>> student = BayesianNetwork([('diff', 'grade'), ('intel', 'grade')])
-        >>> cpd = TabularCPD('grade', 2, [[0.1, 0.9, 0.2, 0.7],
-        ...                               [0.9, 0.1, 0.8, 0.3]],
-        ...                  ['intel', 'diff'], [2, 2])
-        >>> student.add_cpds(cpd)
-        >>> student.get_cpds()
+        >>> from pgmpy.utils import get_example_model
+        >>> model = get_example_model('asia')
+        >>> cpds=model.get_cpds()
```
Comment (ankurankan) [inline pgmpy/models/BayesianNetwork.py:296]: Space before and after `=`. Should be: `cpds = model.get_cpds()`

File: pgmpy/models/BayesianNetwork.py [hunk 8]
```diff
@@ -279,26 +279,45 @@ def get_cpds(self, node=None):
...
+        >>> cpds
+        [<TabularCPD representing P(asia:2) at 0x7dbbd9bdbb80>,
+        <TabularCPD representing P(bronc:2 | smoke:2) at 0x7dbbd9bda3e0>,
+        <TabularCPD representing P(dysp:2 | bronc:2, either:2) at 0x7dbbd9bd8550>,
+        <TabularCPD representing P(either:2 | lung:2, tub:2) at 0x7dbbd9bda800>,
+        <TabularCPD representing P(lung:2 | smoke:2) at 0x7dbbd9bd89d0>,
+        <TabularCPD representing P(smoke:2) at 0x7dbbd9bd8f70>,
+        <TabularCPD representing P(tub:2 | asia:2) at 0x7dbbd9bda860>,
+        <TabularCPD representing P(xray:2 | either:2) at 0x7dbbd9bd9a80>]
+        >>>  cpds = model.get_cpds('bronc')     # Returns the CPD of node
```
Comment (ankurankan) [inline pgmpy/models/BayesianNetwork.py:306]: Change cpds to cpd as we expect the method to only return a single cpd

File: pgmpy/models/BayesianNetwork.py [hunk 9]
```diff
@@ -279,26 +279,45 @@ def get_cpds(self, node=None):
...
+        >>>  cpds
+        <TabularCPD representing P(bronc:2 | smoke:2) at 0x7dbbd9bda3e0>
+        >>> cpds = model.get_cpds('tbu')
```
Comment (ankurankan) [inline pgmpy/models/BayesianNetwork.py:309]: You can remove this example with the error.

**Review by ankurankan — CHANGES_REQUESTED** [review]
I have left a few comments, everything else looks good.

**Comment by ankurankan** [comment]
@B3CODER This PR can be closed. Right?

**Comment by B3CODER** [comment]
Yes sir

**Comment by B3CODER** [comment]
Sir, any issue regarding PR?

**Pull request closed without merging**